src/components/datepicker/MonthPicker.js

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import Box from '../box';
+import { Select } from '../select';
+import { NumericInput } from '../input';
+import theme from './theme.css';
+
+const currentYear = new Date().getFullYear();
+const fromMonth = new Date(currentYear, 0);
+const toMonth = new Date(currentYear + 10, 11);
+
+const MonthPicker = ({ date, localeUtils, onChange, small }) => {
+  const [selectedMonth, setSelectedMonth] = useState({
+    value: date.getMonth(),
+    label: localeUtils.formatMonthTitle(date),
+  });
+  const [selectedYear, setSelectedYear] = useState(date.getFullYear());
+
+  const months = localeUtils.getMonths().map((monthName, index) => {
+    return { value: index, label: monthName };
+  });
+
+  const years = [];
+  for (let i = fromMonth.getFullYear(); i <= toMonth.getFullYear(); i += 1) {
+    years.push(i);
+  }
+
+  useEffect(() => {
+    setSelectedMonth({ value: date.getMonth(), label: localeUtils.formatMonthTitle(date) });
+    setSelectedYear(date.getFullYear());
+  }, [date]);
```

Comment:
I would just use a memoized value instead of using state and and changing with a `useEffect`, to avoid async issues (because useEffect doesn't block a render):

```js
const selectedMonth = useMemo(() => ({ value: date.getMonth(), label: localeUtils.formatMonthTitle(date)}), [date])
const selectedYear = useMemo(() => date.getFullYear(), [date])
```